src/pybind/mgr/dashboard/frontend/src/app/shared/components/select-badges/select-badges.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnChanges } from '@angular/core';
+import { Input } from '@angular/core';
+
+@Component({
+  selector: 'cd-select-badges',
+  templateUrl: './select-badges.component.html',
+  styleUrls: ['./select-badges.component.scss']
+})
+export class SelectBadgesComponent implements OnChanges {
+  @Input() data: Array<string> = [];
+  @Input() options: Array<SelectBadgesOption> = [];
+  @Input() emptyMessage = 'There are no items.';
+
+  constructor() {}
+
+  ngOnChanges() {
+    if (this.options) {
+      this.options.forEach((option) => {
+        if (this.data.indexOf(option.name) !== -1) {
+          option.selected = true;
+        }
+      });
+    }
+  }
+
+  private updateOptions() {
+    this.data.splice(0, this.data.length);
```

Comment:
This function could be refactored. Here my diff:
```diff
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/shared/components/select-badges/select-badges.component.ts b/src/pybind/mgr/dashboard/frontend/src/app/shared/components/select-badges/select-badges.component.ts
index 01d0904..d6a626a 100644
--- a/src/pybind/mgr/dashboard/frontend/src/app/shared/components/select-badges/select-badges.component.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/shared/components/select-badges/select-badges.component.ts
@@ -14,7 +14,7 @@ export class SelectBadgesComponent implements OnChanges {
   constructor() {}
 
   ngOnChanges() {
-    if (this.options) {
+    if (this.data.length > 0) {
       this.options.forEach((option) => {
         if (this.data.indexOf(option.name) !== -1) {
           option.selected = true;
@@ -24,12 +24,7 @@ export class SelectBadgesComponent implements OnChanges {
   }
 
   private updateOptions() {
-    this.data.splice(0, this.data.length);
-    this.options.forEach((option: SelectBadgesOption) => {
-      if (option.selected) {
-        this.data.push(option.name);
-      }
-    });
+    this.data = this.options.filter(option => option.selected).map(option => option.name);
   }
 
   selectOption(option: SelectBadgesOption) {

```

Reply:
`this.data = ...` will not work here because we cannot assign a new reference to a field that is an `@Input` of the component.

Reply:
Ok, than this function is fine with me, but could you change line 17 as the if clause will always resolve to true unless you give it a `undefined` input. IMO `this.data.length > 0` would make more sense there.